app/qml/components/DateTimePicker.qml

```diff
@@ -0,0 +1,628 @@
+import QtQuick 2.14
+import QtQuick.Layouts 1.14
+import QtQuick.Controls 2.14
+import Qt.labs.calendar 1.0
+import QtGraphicalEffects 1.14
+
+// Inspired by https://gist.github.com/ilnuribat/5cc6010aea7e35708ff90c65d7be5bcf
+
+Item {
+  id: root
+
+  property bool hasTimePicker: true
+  property bool hasDatePicker: true
+
+  property date dateToSelect: new Date()
+
+  property int fontPointSizeBig: 18
+  property int fontPointSizeNormal: 15
+  property int fontPointSizeSmall: 12
+
+  signal selected(date selectedDate)
+  signal canceled()
+
+  function getSelectedTime() {
+    if (hasDatePicker && hasTimePicker) {
+      let ddate = calendar.selectedDate
+      let time = timepicker.time
+      var newDate = new Date(
+            ddate.getFullYear(),
+            ddate.getMonth(),
+            ddate.getDate(),
+            time.getHours(),
+            time.getMinutes(),
+            time.getSeconds(),
+            time.getMilliseconds()
+            )
+      return newDate
+    } else if (hasDatePicker) {
+      return calendar.selectedDate
+    } else if (hasTimePicker) {
+      return timepicker.time
+    }
+    return new Date()
+  }
+
+  function setDate(toDate) {
+    calendar.selectDate(toDate)
+    calendar.navigateToDate(toDate)
+    timepicker.setTime(toDate)
+  }
+
+  onDateToSelectChanged: {
+    setDate(dateToSelect)
+  }
+
+  focus: true
+
+  Keys.onReleased: {
+    if ( event.key === Qt.Key_Back || event.key === Qt.Key_Escape ) {
+      event.accepted = true
+      root.canceled()
+    }
+  }
+
+  height: 400
+  width: 400
+
+  QtObject {
+    id: palette
```

Comment:
I would prefer to use white color for back and save, or do we have this orange on green somewhere else?

Reply:
I am not sure about this, I wanted to let users know that these areas are clickable